### PR TITLE
Biomart mappings testable by species and less of Atlas Prod env var

### PR DESCRIPTION
**Pull request by pcm32**
This PR is a bit more varied than it should be, but this is partly because when I started I wasn't sure what was required to be done to achieve what I wanted:

- Make GO Depths optional (default to be done) to avoid connections to Oracle db when testing. In time we will deprecate GO depths as it is conceptually incorrect. I could revert this here and cherry pick it into a new PR if needed, as this is the biggest thing unrelated to the rest of the PR.
- Biomart mappings testable individually per organism. This is achieved by: 
  - Making the path to use for annsrcs configurable, and then pointing to an external folder that contains only the desired organism files. This also opens the door for Ensembl updates limited to certain species.
  - Adding a @Main annotation to retrieve.sc so that only the validation step is executed.
  - Although external to this PR, additions to atlas-embassy-infrastructure on the Jenkins part to make use of this.
  - Adds a dockerfile, built by https://quay.io/repository/ebigxa/ensembl-update-env for the ammonite/scala environment needed for running this on Kubernetes through Jenkins.
- Where possible, substitution of the ATLAS_PROD env variable by more specific path variables. In general our modules should be independent of the choice of paths that we make in the file system, which facilitates testing outside of the EBI cluster and is more specific to the developer in what the heck is needed in terms of files (ATLAS_PROD is unfortunately a very populated directory). This will be continue to be done progressively until atlas-annotation can run completely independently of the EBI cluster setup.
- As with species, the experiment directories is also made configurable, so that this can be applied to specific experiments if needed.

### CODE REVIEW

File: src/Directories.sc [hunk 1]
```diff
@@ -27,18 +33,55 @@ lazy val alternativeToCanonicalGoTermMapping = {
...
 val annsrcsPath = PROJECT_ROOT/"annsrcs"/"ensembl"
 val wbpsAnnsrcsPath = PROJECT_ROOT/"annsrcs"/"wbps"
 
+lazy val ANNOTATION_SOURCES: Seq[Path] = Option(System.getenv.get("ANNOTATION_SOURCES"))
+  .map(_.split(":").map(Path(_)).filter(exists).filter(_.isDir))
+match {
+  case Some(paths) => ((paths.map(ls! _).flatten) ++ (List())).toList
+  case None => ((ls! wbpsAnnsrcsPath) ++ (ls! annsrcsPath))
+}
+
```
[the remark this answers is not in the document]
Comment (pcm32) [reply src/Directories.sc]: Sure, will add something along those lines, thanks!

File: src/atlas/AtlasSpecies.sc [hunk 2]
```diff
@@ -49,7 +45,7 @@ object AtlasSpeciesFactory {
           "metazoa" -> List(("Ensembl Genomes", "https://metazoa.ensembl.org/")),
           "fungi" -> List(("Ensembl Genomes", "https://fungi.ensembl.org/")),
           "parasite" -> List(("Wormbase ParaSite", "https://parasite.wormbase.org/")),
-          "plants" -> List(("Gramene", "https://ensembl.gramene.org/"),("Ensembl Genomes", "https://plants.ensembl.org/")),
+          "plants" -> List(("Gramene", "http://ensembl.gramene.org/"),("Ensembl Genomes", "https://plants.ensembl.org/")),
           "protists" -> List(("Ensembl Genomes", "https://protists.ensembl.org/"))
         )
```
[the remark this answers is not in the document]
Comment (pcm32) [reply src/atlas/AtlasSpecies.sc]: If you see the original PR from Alfonso #5, he mentions that gramene doesn't support https yet.

File: src/Directories.sc (hunk 1 above)
[the remark this answers is not in the document]
Comment (pcm32) [reply src/Directories.sc]: There!

[the remark this answers is not in the document]
Comment (suhaibMo) [reply src/Directories.sc]: @pcm32 Thanks !

**Pull request merged**